ElementX/Sources/AccessibilityTests/AccessibilityTestsAppCoordinator.swift

```diff
@@ -0,0 +1,152 @@
+        UIView.setAnimationsEnabled(false)
+        
+        MXLog.configure(currentTarget: "accessibility-tests")
+        
+        ServiceLocator.shared.register(userIndicatorController: UserIndicatorController())
+        
+        AppSettings.configureWithSuiteName("io.element.elementx.accessibilitytests")
+        AppSettings.resetAllSettings()
+        ServiceLocator.shared.register(appSettings: AppSettings())
+        ServiceLocator.shared.register(bugReportService: BugReportServiceMock(.init()))
+        
+        let analyticsClient = AnalyticsClientMock()
+        analyticsClient.isRunning = false
+        ServiceLocator.shared.register(analytics: AnalyticsService(client: analyticsClient,
+                                                                   appSettings: ServiceLocator.shared.settings))
+        
+        guard let name = ProcessInfo.accessibilityViewID,
+              let previewType = TestablePreviewsDictionary.dictionary[name]
+        else { fatalError("Unable to launch with unknown screen.")
+        }
+        previewsWrapper = .init(name: name, previews: previewType._allPreviews)
+        
+        setupSignalling()
+    }
+    
+    func toPresentable() -> AnyView {
+        AnyView(PreviewsWrapperView(wrapper: previewsWrapper))
+    }
+    
+    private func setupSignalling() {
+        do {
+            let client = try UITestsSignalling.Client(mode: .app)
+            client.signals.sink { [weak self] signal in
+                guard let self else { return }
+                switch signal {
+                case .nextPreview:
+                    Task { [weak self] in
+                        guard let self else { return }
+                        await previewsWrapper.updateCurrentIndex()
+                        do {
+                            guard !previewsWrapper.isDone else {
+                                try client.send(.noMorePreviews)
+                                return
+                            }
+                            
+                            try client.send(.nextPreviewReady(name: previewsWrapper.previewName))
+                        } catch {
+                            fatalError("failed sending signal: \(signal)")
+                        }
+                    }
+                default:
+                    break
+                }
+            }
+            .store(in: &cancellables)
+        } catch {
+            fatalError("Unable to start client signalling")
+        }
+    }
+}
+
+struct PreviewsWrapperView: View {
+    @ObservedObject var wrapper: PreviewsWrapper
+    
+    var body: some View {
+        if wrapper.currentIndex < 0 || wrapper.isDone {
+            EmptyView()
+        } else {
+            wrapper.currentPreview.content
+                .id(wrapper.previewName)
+        }
+    }
+}
+
+final class PreviewsWrapper: ObservableObject {
+    private let name: String
+    private let previews: [_Preview]
+    @Published var currentIndex = -1
```

Comment:
Can we try this is observation instead?

```suggestion
@Observable final class PreviewsWrapper {
    private let name: String
    private let previews: [_Preview]
    private(set) var currentIndex = -1
```